docs/developer-guide/ksqldb-clients/contributing.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: page
+title: Contributing a new ksqlDB client
+tagline: Contributing a new ksqlDB client
+description: Contribute a new ksqlDB client for your favorite language.
+keywords: ksqlDB, client
+---
+
+This page contains information and recommendations for how to contribute a new [ksqlDB client](index.md).
+
+Functionality
+-------------
+
+Clients should support the following functionality:
+- [Push and pull queries](#push-and-pull-queries)
+- [Terminating push queries](#terminating-push-queries)
+- [Inserting new rows of data into existing ksqlDB streams](#inserting-new-rows-of-data-into-existing-ksqldb-streams)
+- [Creation of new streams, tables, and persistent queries](#creation-of-new-streams-tables-and-persistent-queries)
+- [Terminating persistent queries](#terminating-persistent-queries)
+- [Listing existing streams, tables, topics, and queries](#listing-existing-streams-tables-topics-and-queries)
+
+### Push and pull queries ###
+
+Push and pull queries may be issued by sending requests to the [`/query-stream` server endpoint](TODO).
+If no errors are encountered while processing the query statement, the server will write a
```

Comment:
I can parse what kind of response I should expect from the server, but I don't understand what I need to supply in the request. It might be good to divide these categories up into subsections and make tables out of them. This is the kind of information that's really important to be able to correctly reference as you work.

Reply:
Ditto with the rest of the sections. Im not sure what I should really expect to supply in the request or expect in the response. If all of that will show in the server docs, we should note that every time.

Reply:
Good call. I've explicitly called out that implementers should reference the server docs in each section, and added information in places where more details were needed.